sellotape/sellotape_dj/main_app/views.py

```diff
@@ -1,4 +1,45 @@
 from django.shortcuts import render
+import datetime
+from.models import Stream, User
+# pip install --upgrade google-api-python-client
+from apiclient.discovery import build
+
+api_key = "AIzaSyCAL9HUYTpY9vX05h2Wnk5CAmYmPvSxX8E"
+youtube = build('youtube', 'v3', developerKey=api_key)
 
 def index(request):
     return render(request, 'sellotape/index.html')
+	
+def get_channel_videos(request):
+    #shuold be added to User
+    channel_id = request.user.get_youtube_link()
+    # get Uploads playlist id
+    res = youtube.channels().list(id=channel_id, part='contentDetails').execute()
+    playlist_id = res['items'][0]['contentDetails']['relatedPlaylists']['uploads']
+    videos = []
+    next_page_token = None
+
+    while true:
+        res = youtube.playlistItems().list(playlistId=playlist_id,
```

Comment:
view functions should generally return very quickly because the end user's browser is waiting for a response. OT seems to me that this does a REST API call to YouTube, so it would probably not return very quickly or may not return at all if YouTube is down for some reason.

On other words, I think this code belongs is some background processing function or a stand-alone script rather then a view function.